src/adjustproxyconnection.cpp

```diff
@@ -0,0 +1,262 @@
+      logger.debug() << "Unknown ProcessingState: " << m_state;
+  }
+}
+
+void AdjustProxyConnection::processFirstLine() {
+  if (m_buffer.isEmpty()) {
+    return;
+  }
+
+  int pos = m_buffer.indexOf("\n");
+  if (pos == -1) {
+    return;
+  }
+
+  QByteArray line = m_buffer.left(pos);
+  m_buffer.remove(0, pos + 1);
+
+  QList<QByteArray> parts = line.split(' ');
+  if (parts.length() < 2) {
+    logger.error() << "Invalid HTTP request";
+    m_connection->close();
+    return;
+  }
+
+  m_method.append(parts[0]);
+  m_route = parts[1].trimmed();
+
+  m_state = ProcessingState::FirstLineDone;
+}
+
+void AdjustProxyConnection::processHeaders() {
+  if (m_buffer.isEmpty()) {
+    return;
+  }
+
+  while (true) {
+    int pos = m_buffer.indexOf("\n");
+    if (pos == -1) {
+      return;
+    }
+
+    QByteArray line = m_buffer.left(pos);
+    m_buffer.remove(0, pos + 1);
+
+    QByteArray header = line.trimmed();
+    if (header.isEmpty()) {
+      break;
+    }
+
+    pos = header.indexOf(":");
+    if (pos == -1) {
+      continue;
+    }
+
+    QByteArray headerName = header.left(pos);
+    header.remove(0, pos + 1);
+    QByteArray headerValue = header.trimmed();
+
+    QPair<QString, QString> headerPair;
+    headerPair.first = QString(headerName);
+    headerPair.second = QString(headerValue);
+
+    if (QString::compare(headerPair.first, "host", Qt::CaseInsensitive) == 0) {
+      continue;
+    }
+
+    if (QString::compare(headerPair.first, "content-length",
+                         Qt::CaseInsensitive) == 0) {
+      bool ok;
+      m_contentLength = headerPair.second.toInt(&ok, 10);
+      if (!ok) {
+        logger.error() << "Content Length could not be parsed";
+        m_connection->close();
+        return;
+      }
+      continue;
+    }
+
+    m_headers.append(headerPair);
+  }
+
+  m_state = ProcessingState::HeadersDone;
+}
+
+void AdjustProxyConnection::processParameters() {
+  if (m_buffer.trimmed().length() > m_contentLength) {
+    logger.error() << "Buffer longer than the declared Contend-Length";
+    m_connection->close();
+    return;
+  }
+
+  if (m_method == "GET") {
+    m_parameters = QUrlQuery(m_route);
+  } else {
+    if (m_buffer.trimmed().length() < m_contentLength) {
+      return;
+    }
+    m_parameters = QUrlQuery(m_buffer.trimmed());
+  }
+
+  m_state = ProcessingState::ParametersDone;
+}
+
+void AdjustProxyConnection::filterParametersAndForwardRequest() {
+  QList<QPair<QString, QString>> newParameters;
+  QList<QPair<QString, QString>> unknownParameters;
+
+  for (QPair<QString, QString> parameter : m_parameters.queryItems()) {
+    if (allowList.contains(parameter.first)) {
+      newParameters.append(
+          QPair<QString, QString>(parameter.first, parameter.second));
+    } else if (denyList.contains(parameter.first)) {
+      newParameters.append(
+          QPair<QString, QString>(parameter.first, denyList[parameter.first]));
+    } else {
+      unknownParameters.append(
```

Comment:
Let's still add it to the newParameters and let's store the parameter.first in the unknownParameters. The unknownParameters is just for reporting.